apps/src/lib/kits/maker/boards/microBit/LightSensor.js

```diff
@@ -86,13 +86,11 @@ export default class LightSensor extends EventEmitter {
 
   // Reset the state to initial values between runs
   reset() {
-    this.state = {
-      threshold: 128,
-      rangeMin: 0,
-      rangeMax: MAX_LIGHT_SENSOR_VALUE,
-      currentReading: 0,
-      currentBufferWriteIndex: 0
-    };
+    this.state.threshold = 128;
```

Comment:
This is subtle and interesting.  Maybe a comment mentioning that `this.state` already has existing values that you don't want to touch, for future visitors to the code?